### DISCUSSION TITLE
기술 용어 번역 표준화를 위한 안내서

### DISCUSSION THREAD

**Discussion by hochan222** [ko-translation-team/glossary#1] [공지사항]
안녕하세요. 🙇🙇

기술 용어 번역에 대해 개인 혹은 집단마다 다른 용어를 사용하는 경우가 많습니다. 이러한 혼란을 해결하고 기술 용어의 표준화와 그 과정을 기록하기 위해, 기술 용어에 대한 번역 안내서를 만들고자 현재 저장소를 @alicefromkr 님, @XiNiHa 님, @BasixKOR 님과의 디스코드를 통해 만들게 되었습니다.

다음은 기술 용어 안내서를 작성하기 전에 고려해야 할 사항들 입니다. 논의되면 좋을 주제들 별도의 댓글로 남겨주시면 본문에 추가하겠습니다.

### 기준

표준화 과정에서 기준을 마련하는 일은 생각보다 굉장히 복잡하고 어렵습니다. 사람마다 용어에 대해 어색한 정도가 다르기도하고, 올바른 표기가 항상 대중적인 표기는 아니기 때문입니다. 따라서, 이에 대해 최소한의 방향성을 정할 필요가 있습니다.

Javascript의 **Promise**는 대표적인 예시입니다. 다음 세 가지로 번역할 수 있습니다.

(1) 프로미스
(2) 프라미스
(3) promise

Promise의 미국 발음은 "프라미스"입니다. 하지만 한국에서는 프로미스가 더 익숙합니다. 실제 구글 결과도 100배 이상 프로미스의 대중적인 인지도가 높습니다. 이런 이유에서인지 책에서도 여러가지 표기가 사용되고 있습니다.

**"프라미스"라는 명칭을 사용하는 번역 자료들**(함수형 자바스크립트 입문 2/e(도서), 자바스크립트 코딩의 기술(도서), 모던 자바스크립트 튜토리얼(웹)), **"프로미스"라는 명칭을 사용하는 번역 자료들**(O'RELLY 타입스크립트 프로그래밍(도서))

Promise 영어 그대로 표기하자는 관점도 있을 수 있습니다. 하지만, 원문에서 promise가 소문자로 사용될 경우 번역 시 소문자로 표기해야하는지의 문제가 있습니다. 또한, 어느 범위까지 영어 단어 그대로 표기해야할지의 문제도 있습니다. Event Handler, Typescript도 영어 단어 그대로 표기해야할까요? 

> 참고: Event Handler를 React 공식 문서에서는 "이벤트 핸들러"로 MDN 이나 Microsoft 문서에서는 "이벤트 처리기"로 번역하고 있습니다.

> 참고: MDN ko 커뮤니티는 대체할만한 한국어가 있을 경우 최대한 번역하는 것을 지향합니다. 예를들어, 희소 배열(Sparse array), 범위(Scope), 접근(Access), 형식화 배열(TypedArray)이 있습니다. 대체 불가능 할 경우 한국어 표기를 지향합니다. 예를들어,쉐도우돔, 프로토타입, 클래스가 있습니다. 하지만, Promise chaining의 경우 "프로미스 연결하기"가 아닌 "프로미스 체이닝"으로 번역이 결정된 사례도 있습니다.

### 범위

저장소에서 논의할 기술 용어 범위를 정하는 것이 좋습니다. C++, Java, Rust등 각 언어에 따라 기술 용어 번역 차이가 존재합니다. **"모든 IT 기술 용어"** 로 범위를 정할지와 각각 페이지를 분리하는것이 좋을지 고민이됩니다.

### 안내서 형식

안내서에 필수적으로 들어가야하는 요소들과 형식에 대한 내용입니다. 

### PR 병합 방식과 조직 맴버 권한 및 가입 조건

최소 몇명의 검토를 통해 병합할 것인지와 조직의 가입을 자유롭게 할 것인지와 병합 권한을 어떻게 유지할지에 대한 내용입니다.

---

우선 당장 떠오르는 고민해보면 좋을 내용들을 적어 보았습니다. 아직 아무런 틀이 정해지지 않은 만큼 다양한 의견 있으면 말씀해주세요!

감사합니다! 🙇🙇

**Comment by alicefromkr** [comment]
안녕하세요! 

### 안내서 형식 
번역 형식에 고민을 해봤는데요, @RanolP 님께서 비슷하게 [ko-translation-terms](https://github.com/RanolP/ko-translation-terms)를 만드신게 있습니다. 

|원어|음차|번역어|
|---|----|------|

이렇게 단어와 발음 그리고 한국어 번역을 올려두면 좋을 것 같습니다. 

저는 여기에 따로 **비고**란을 추가해서 어디서 참고를 해왔는지 혹은 어떤 토론을 진행했는지에 대한 링크를 추가하면 좋을 것 같습니다. 

**예시로**
|원어|음차|번역어|비고|
|---|----|------|----|
|Promise|프로미스| | [음차 토론](https://github.com/ko-translation-team/glossary/discussions/1)|
|Footer|푸터|바닥글|-|
|Protocol|프로토콜|통신규약|[다듬은말](https://www.korean.go.kr/front/imprv/refineView.do?mn_id=158&imprv_refine_seq=12304)|

### 범위
저는 전체 범위인 "모든 IT 기술 범위" 보다는 분야별로 나누는게 좋을 것 같습니다. 이 부분은 다른 분들 의견도 함께하면 좋을 것 같습니다. 

### 조직 맴버 권한 및 가입 조건
저는 자유롭게 가입하는게 좋다고 생각하는데요, 그렇게 되면 관리가 어렵게 되겟죠..? 이부분은 확실히 정하기 어렵네요.

**Reply by alicefromkr** [reply]
아 그리고 원활한 커뮤니케이션을 위해서 디스코드를 하나 만들어뒀습니다! 
[디스코드 초대 링크](https://discord.gg/rSNarpnr)

**Reply by RanolP** [reply]
저는 당시 유지보수 비용을 최소화하고 싶었어서 기능을 꽤나 누락시켰었는데 이광근 교수님을 필두로 운영되는 https://easyword.kr/ 사이트처럼 누구나 제안할 수 있되, 거기에 더해 합의를 통해 토의록과 함께 특정 어휘를 승인하는 형태가 좋지 않나 생각합니다.
또한 같은 말에서 유래해 품사가 다른 말(abstraction: 추상화, abstract: 추상적인, 추상화하다)나 비슷한 말/반댓말을 그래프 형태로 볼 수 있는 시스템이 있으면 좋을 것 같습니당

**Reply by hochan222** [reply]
@RanolP 님, 오! 넘넘 좋은 사이트네요. 최대한 활용하면 좋을듯합니다..!

> 이렇게 단어와 발음 그리고 한국어 번역을 올려두면 좋을 것 같습니다.

@alicefromkr 님, 형식 넘넘 좋습니다. 한가지 우려되는 점은 현재 저장소의 목적 중 하나는 개인 혹은 집단이 사용하는 기술 용어를 통일하는 것에 있다고 생각하는데요..! 음차와 번역어 둘 다 안내서에 포함시키면, 음차 용어와 번역어를 혼용하는 경우 있지 않을까 걱정됩니다. 따라서, 원어에 대해 대응되는 단어는 번역어 한개가 좋지 않을까 의견드립니다. 

용어가 사용되는 영역에 따라 의미가 달라질 수 있을 듯하여, 사용되는 영역에 대한 정보도 포함시키면 좋을것같아요. 예를들어, module의 경우 "ESM에서 명시하는 모듈"과 "어떠한 교육 과정, 단위에 의한 표현의 '과정'"이 있을 수 있습니다.

음차를 작성해서 얻는 장점에는 어떤것이 있을까요?

**Reply by RanolP** [reply]
easyword.kr 자체를 활용하기엔 저쪽 성향이 살짝 최대한 외래어/한자어를 피한다 쪽이라서 언어의 사회성 면에서 살짝 걱정이 됩니다. 과도하게 순화시킨 단어로 인지될 만한 어휘가 많다보니(국립국어원의 다듬은 말이 흔히 북한말이냐고 이야기 듣듯이요 😅)... 형식만 빌리는 편이 좋다고 생각합니다.

음차한 어휘 자체를 사용하길 권장한다기보단 한국어->영어로 역방향 탐색에 쓸 목적을 갖고 있었습니다. 가령 수정 제안을 하는 VS Code 확장을 만든다든지 할 때 데이터로 사용한다면 좋지 않을까요?! 마찬가지의 이유로 권장하는 어휘 하나만 두기보단 이런 단어와 같은 말로도 사용된다라는 등의 목록도 제공하면 좋을 것 같습니다

**Reply by RanolP** [reply]
사용 영역에 대한 정보 역시 포함하면 좋을 듯 한데, 기존 사전들이 하는 접근처럼 abstract[1] 간략한 의미, 번역어, 예문 / abstract[2] 간략한 의미, 번역어, 예문 이런 식으로 넣으면 어떨까용

**Reply by hochan222** [reply]
몬가몬가 같이 활용하면 좋을듯한데, 배경/원칙에 "이때, 어깨에 힘을 뺀다. 지레 겁먹게하는 용어(불필요한 한문투)를 피하고, 가능하면 쉬운말을 찾는다." 문장이 있어서 말씀 주신대로 조금 어려워보이네요. ㅜㅜ

주신 의견 모두 좋습니다~! 👍👍

**Reply by roo0t** [reply]
안녕하세요 :)

논의를 거쳐 나온 최종적인 추천(?) 번역어는 표로 깔끔하게 정리하되, 각 용어에 대해서는 별도의 섹션이나 문서로 논의 과정이나 각 번역어에 대한 pros/cons 등의 정보가 제공되면 좋을 것 같습니다. 그리고 번역어마다 참조할 수 있는 실제 사용례(공식 문서라든지, 위키피디아, 도서 등)를 정리해 두면 해당 번역어가 얼마나 널리 쓰이는지, 그리고 실제 문장 속에서 얼마나 어색하게/자연스럽게 보이는지를 알기 좋을 것 같습니다.

**Comment by hochan222** [comment]
## 보편적인 기술 용어

`배열(Array)`, `오류(Error)`와 같이 보편적으로 받아들여져서 토론할 필요가 없는 용어들을 먼저 모아서 기술 용어 안내서를 작성하는 건 어떨까요? 

조금이라도 논란의 여지가 있는 용어들은 따로 모아두고 절차를 걸쳐 정하는 것도 좋을듯 합니다..! 

| 원어 | 번역어 |
| ---- | ---- |
| Array | 배열 |
| Error | 오류 |

**Comment by hochan222** [comment]
## 조직 맴버 권한 및 가입 조건

우선, owner 권한 모두 드렸습니다..! (cc. @alicefromkr 님, @RanolP 님, @BasixKOR 님, @XiNiHa 님)

[ko-translation-team](https://github.com/ko-translation-team) 조직은 모두에게 열어두고, [glossary 저장소](https://github.com/ko-translation-team/glossary)만 PR 리뷰 및 병합하는 기여자 외의 레포 맴버 등록을 하지 않는 방향은 어떤가요? 

다음과 같은 이점이 있습니다.

- 모든 인원이 ko-translation-team 조직 내에서 자유롭게 활동 가능하다.
- glossary 저장소 권한 관리가 가능하다.

의견 부탁드립니다 🙇🙇

---

동의하신다면... 매번 모든 분께 초대를 드릴 수 없기에, 조촐하지만 ko-translation-team 조직 자동 초대장 사이트를 만들었습니다. 동작 확인하긴 했는데 테스트 한번 부탁드립니다..!

- github: https://github.com/ko-translation-team/invitation
- url: https://ko-translation-team-hochan222.koyeb.app/

**Reply by XiNiHa** [reply]
얼핏 듣기에는 조직 가입을 위한 페이지를 운영하는 데에 불필요하게 리소스가 소모될 것처럼 느껴지는데 반해, 장점이 그다지 와닿지 않는 것 같습니다. 개인적으로는 glossary 저장소의 PR 머지 권한이 부여되는 것이 아니라면 사람들이 조직에 가입할 동기가 딱히 없지 않을까 하는 생각이 드네요 🤔

**Reply by hochan222** [reply]
조직에 가입하는 것만으로도 관심과 잠재적인 기여 의사를 표현할 수 있고 이는 추후 기여의 주요 동기, 홍보, 신뢰성 지표에 긍정적 영향을 줄 수 있다고 생각합니다.

하지만 페이지 운영에 리소스가 드는건 사실이기에 생각보다 효과가 미미하거나 와닿지 않는다면 pr 머지 권한 있는 분들만 가입하는 방향이 좋을듯합니다..!

**Reply by hochan222** [reply]
요건 빨리 의사결정이 되면 좋을듯하여 멘션드립니다. 의견부탁드려요! cc. @BasixKOR 님, @alicefromkr 님, @roo0t 님, @RanolP 님

다른분들도 의견 부탁드립니다!!

**Reply by alicefromkr** [reply]
저도 pr 머지 권한 있는 분들만 가입하는 방향이 좋을것 같습니다!
